core/Binary.carp

```diff
@@ -0,0 +1,101 @@
+(system-include "carp_binary.h")
+
+(defmodule Binary
+  (doc Order 
+    "The type of byte orders. 
+
+    LittleEndian designates the little endian ordering, and indicates the least
+    significant byte appears first in a given byte sequence.
+
+    BigEndian designates the big endian ordering, and indicates the most
+    significant byte occurs first in a given byte sequence.")
+  (deftype Order LittleEndian BigEndian)
```

Comment:
How about adding a function that returns the host system’s endianness?

Reply:
Great idea, thanks!